docs/user/troubleshootingbootstrap.md

```diff
@@ -0,0 +1,220 @@
+
+1. Use the user's already setup `SSH_AGENT`. If the user has a ssh-agent setup, the installer will use it for SSH authentication.
+
+2. Use the user'd home directory, `~/.ssh` on linux hosts, to load all the SSH private keys and use those for SSH authentication.
+    a. The installer also configures the bootstrap host with a *generated* SSH key, and this private key will be used for SSH authentication none of the user keys are trusted.
+    The installer only configures the bootstrap host to trust the generated key, and therefore the log bundle will only contain the logs from bootstrap host and not the control-plane hosts.
+
+### Using the user provisioned workflow
+
+When users are creating the infrastructure for the OpenShift cluster and the cluster fails to bootstrap, the users can use the `gather bootstrap` subcommand to gather the logs from the bootstrap host.
+
+```console
+$ openshift-install gather bootstrap --help
+Gather debugging data for a failing-to-bootstrap control plane
+
+Usage:
+  openshift-install gather bootstrap [flags]
+
+Flags:
+      --bootstrap string     Hostname or IP of the bootstrap host
+  -h, --help                 help for bootstrap
+      --key stringArray      Path to SSH private keys that should be used for authentication. If no key was provided, SSH private keys from user's environment will be used
+      --master stringArray   Hostnames or IPs of all control plane hosts
+```
+
+An example of a invocation for a cluster with three control-plane machines would be,
+
+```sh
+openshift-install gather bootstrap --bootstrap ${BOOTSTRAP_HOST_IP} --master ${CONTROL_PLANE_1_HOST_IP} --master ${CONTROL_PLANE_2_HOST_IP} --master ${CONTROL_PLANE_3_HOST_IP}
+```
+
+#### Authenticating with bootstrap host for upi
+
+When explicitly using the `gather bootstrap` subcommand, user can either utilize installer's discovery mechanism like detailed [above](#authenticating-with bootstrap host-for-ipi) or you can provide the keys using `--key` flag.
+
+An example of a invocation for a cluster with three control-plane machines would be,
+
+```sh
+openshift-install gather bootstrap --key ${KEY_1} --key ${KEY_2} --bootstrap ${BOOTSTRAP_HOST_IP} --master ${CONTROL_PLANE_1_HOST_IP} --master ${CONTROL_PLANE_2_HOST_IP} --master ${CONTROL_PLANE_3_HOST_IP}
+```
+
+## Understanding the bootstrap failure log bundle
+
+Here's what a log bundle looks like,
+
+```console
+.
+├── bootstrap
+├── control-plane
+├── failed-units.txt
+├── rendered-assets
+├── resources
+└── unit-status
+
+5 directories, 1 file
+```
+
+### file: failed-units.txt
+
+The failed-units.txt contains a list of all the **failed** systemd units on the bootstrap host.
+
+### directory: unit-status
+
+The unit-status directory contains the details of each failed systemd unit from [failed-units][#file-failed-units-txt]
```

Comment:
nit: `[failed-units](#file-failed-units-txt).` (braces -> parens and trailing period).